esiclient/tests/functional/v1/test_image.py

```diff
@@ -0,0 +1,168 @@
+#    Copyright (c) 2016 Mirantis, Inc.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+
+import ddt
+from tempest.lib import exceptions
+
+from esiclient.tests.functional import base
+from esiclient.tests.functional import utils
+
+
+@ddt.ddt
+class ImageTests(base.ESIBaseTestClass):
+    """Functional tests for ESI network commands."""
+
+    @classmethod
+    def setUpClass(cls):
+        super(ImageTests, cls).setUpClass()
+        cls._init_dummy_project(cls, 'project1', 'member')
+        cls._init_dummy_project(cls, 'project2', 'member')
+
+    def setUp(self):
+        super(ImageTests, self).setUp()
+        self.clients = ImageTests.clients
+        self.users = ImageTests.users
+        self.projects = ImageTests.projects
+        if 'image_path' not in ImageTests.config['functional']:
+            self.fail('image_path must be specified in test config')
+        self.image_path = ImageTests.config['functional']['image_path']
+
+    def test_admin_image_public_and_private(self):
+        """Tests that an admin can create a public image and set it to private
+
+        Test steps:
+        1) Admin creates public image.
+        2) User 1 can view image.
+        3) Admin sets image to private.
+        4) User 1 can no longer view image.
+        5) Admin deletes image.
+        6) (cleanup) Delete the image created in step 1.
+
+        """
+
+        image = utils.image_create(self.clients['admin'], self.image_path)
+        self.addCleanup(utils.image_delete,
+                        self.clients['admin'],
+                        image['name'],
+                        fail_ok=True)
+
+        image_show = utils.image_show(self.clients['project1-member'],
+                                      image['name'])
+        self.assertEqual(image['id'], image_show['id'])
+
+        utils.image_set(self.clients['admin'],
+                        image['name'], '--private')
+        self.assertRaises(exceptions.CommandFailed,
+                          utils.image_show,
+                          self.clients['project1-member'],
+                          image['name'])
+
+        utils.image_delete(self.clients['admin'], image['name'])
```

Comment:
I still don't get it. For the tests below, project1 creates the image and it should be able to delete that image?

Reply:
I guess the cleanup would fail if it didnt have the fail_ok on it. Its a bit weird we allow tests in general to have their cleanup fail. Admittedly I did not think about this in my code in my pr

Reply:
I'd argue that specifying "fail_ok" means that we are explicitly not allowing the cleanup to fail :)

Reply:
but you set fail_ok to True?

Reply:
Yep, that's definitely possible - but the test will still fail, and presumably the person running the test would have to investigate anyway!

Reply:
I think that's impossible, simply because the cleanup step itself can fail.

I do think that the cleanup steps *should* attempt to undo whatever happens during the test though.

Reply:
Are my comments in the least bit something worth talking about or am I just stalling you merging this pr and going about your weekend?

Reply:
Well, the trunk port tests create their own networks (and delete them after), but if you look at the node_network tests, we attach networks to a node, and then delete the networks, and then undeploy the node (which cleans the node and unattaches networks).

These are fine questions! Don't worry, if it were time for me to log off I would be logged off :)

Reply:
Im just making sure our methodology for failing is ok. I think things are fine though